src/promptflow-evals/promptflow/evals/evaluators/_content_safety/flow/evaluate_with_rai_service.py

```diff
@@ -185,13 +194,15 @@ def evaluate_with_rai_service(
     if credential is None or credential == {}:
         credential = DefaultAzureCredential()
 
+    token = credential.get_token("https://management.azure.com/.default").token
+
     # Get RAI service URL from discovery service and check service availability
-    rai_svc_url = get_rai_svc_url(project_scope, credential)
-    ensure_service_availability(rai_svc_url)
+    rai_svc_url = get_rai_svc_url(project_scope, token)
+    ensure_service_availability(rai_svc_url, token, Tasks.CONTENT_HARM)
 
     # Submit annotation request and fetch result
-    operation_id = submit_request(question, answer, metric_name, rai_svc_url, credential)
-    annotation_response = fetch_result(operation_id, rai_svc_url, credential)
+    operation_id = submit_request(question, answer, metric_name, rai_svc_url, token)
```

Comment:
This could be a bit risky since fetch results can take quite a bit of time sometime and if token expires it will be a bit risky

Reply:
Refined.